bootstrap/pkg/kfapp/coordinator/coordinator.go

```diff
@@ -114,6 +113,10 @@ func getConfigFromCache(pathDir string, kfDef *kfdefsv2.KfDef) ([]byte, error) {
 
 // GetPlatform will return an implementation of kftypes.GetPlatform that matches the platform string
 // It looks for statically compiled-in implementations, otherwise throws unrecognized error
+//
+// TODO(jlewi): Why is platformArgs taken as a []byte? I assume its because each platform might
+// have different types for different platforms. It would probably be better to pass an interface
+// and do type assertion.
```

Comment:
I think I'm going to get rid of it in a follow on PR and just make it part of KfDef.

I'm not seeing an AWS plugin. 
https://github.com/kubeflow/kubeflow/tree/1c07de4d4825f381062c7d7091398b837113b7f1/bootstrap/pkg/kfapp

@Jeffwan Have you started creating the plugin yet?

@kkasravi It looks like #3045 changed getPlatform to no longer call LoadKfApp in the default case. I can change that in this PR. It looks like that also renamed GetKfApp for minikube and gcp to GetPlatform. 

If IUC having unique signatures for minikube and gcp prevents them from being treated as plugins but that should be ok because they are special cased (as is ksonnet).

Reply:
@kkasravi It looks like the issue you are raising wasn't caused by this PR. I created #3520. I'm not sure what the best solution is.

Reply:
Just see this. I am building interface implementation like gcp.go. pkg/kfapp/aws/ @jlewi I have not merged the changes yet.  Still in progress. You can move ahead and I will rebase and make corresponding changes.